pydantic/env_settings.py

```diff
@@ -132,18 +142,16 @@ def __repr__(self) -> str:
 
 
 class EnvSettingsSource:
```

Comment:
I think on reflection, having reviewed the changes to this class, that this feature would better implemented as a separate `Source`. It's actually quite an elegant use of the sources approach.

Reply:
Humm, on reflection, I'm not so sure about this, but we can definitely make the logic clearer.

Reply:
It looks like is better to keep this new logic in one class. If we'll separate it for two `Source`s than we'll need to choose which `Source` to use in the `BaseSettings._build_values` method. Because we cannot use them both at the same time in a lack of efficiency, because they both will be reading the env_file.

Reply:
agreed.